ome/omero-figure
# Open images with

This adds "Open with" support to OMERO.figure.

Uses the `webgateway/open_with/` json data, loading scripts as defined in that data to provide urls for selected image(s) if supported by each Open-with plugin, using the same logic as in https://github.com/openmicroscopy/openmicroscopy/pull/4630.

We always provide the 'Open with: Webclient' link (replaces the previous "Show in Webclient" link). By default the regular webclient 'Image viewer' is also available. In this screen-shot I have added @waxenegger's viewer-ng, allowing you to directly open images in that viewer.

To test:
- configure 'Open with' as described in https://github.com/openmicroscopy/openmicroscopy/pull/4630
- Go to OMERO.figure. Select an image - should see 'Open with' links that are enabled for a single image at top of Info panel (see screenshot).
- If you have configured webclient to 'Open with' OMERO.figure `$ bin/omero config append omero.web.open_with '["omero_figure", "new_figure", {"supported_objects":["images"], "target": "_blank", "label": "OMERO.figure"}]'` this should not show up in figure itself. (Doesn't make sense to "Open with OMERO.figure" when you're already in OMERO.figure!

![screen shot 2016-10-13 at 10 40 36](https://cloud.githubusercontent.com/assets/900055/19344521/6800eab0-9132-11e6-9126-cdce6028c842.png)

NB: this PR is on top of https://github.com/ome/omero-figure/pull/178, which should be merged first.
